### PR TITLE
composer support

### PR DESCRIPTION
**Pull request by candasm**
Hi Alexey,

I have added composer support for yiiframework mobile detect extension

### CODE REVIEW

**Comment by iamsalnikov** [comment]
There is conflict. Please, resolve its and I merge it.

**Comment by candasm** [comment]
I don't know how to do it. I think i need to have a access to your repository.

File: src/MobileDetectComponent.php [hunk 1]
```diff
@@ -1,19 +1,16 @@
 <?php
-
-$basePath = dirname(__FILE__);
-require_once($basePath . DIRECTORY_SEPARATOR . 'lib' . DIRECTORY_SEPARATOR . 'Mobile_Detect.php');
-
 /**
  * Wrapper for Mobile_Detect @link https://github.com/serbanghita/Mobile-Detect
- * @author Alexey Salnikov <me@iamsalnikov.ru>
+ * @author Candas Minareci <candas@minareci.net>
```
Comment (iamsalnikov) [inline src/MobileDetectComponent.php:4]: Why you remove me?

File: src/MobileDetectComponent.php [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
  */
 
-class MobileDetect extends CApplicationComponent
+class MobileDetectComponent extends CApplicationComponent
```
Comment (iamsalnikov) [inline src/MobileDetectComponent.php:7]: Maybe we need to add namespace?

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      "name": "Candasm",
+      "email": "candas@minareci.net"
+    }
+  ],
+  "config": {
+    "vendor-dir": "protected/vendor/"
+  },
+  "autoload": {
+    "psr-4": {
+      "": "src/"
```
Comment (iamsalnikov) [inline composer.json:17]: maybe we need to add some namespace root?

File: README.md [hunk 4]
```diff
@@ -8,7 +7,7 @@ In app config:
     ...
 
     'mobileDetect' => array(
-        'class' => 'ext.MobileDetect.MobileDetect'
+        'class' => 'application.vendors.candasm.yii1-mobile-detect-component.src.MobileDetectComponent'
```
Comment (iamsalnikov) [inline README.md:10]: I think, that there should be class (for example: `iamsalnikov\MobileDetect\MobileDetect`)

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+  },
+  "homepage": "https://github.com/candasm/MobileDetect",
+  "require": {
+    "yiisoft/yii": "1.1.14",
+    "mobiledetect/mobiledetectlib": "~2.7.6"
```
Comment (iamsalnikov) [inline composer.json:23]: "mobiledetect/mobiledetectlib": "*"

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    "vendor-dir": "protected/vendor/"
+  },
+  "autoload": {
+    "psr-4": {
+      "": "src/"
+    }
+  },
+  "homepage": "https://github.com/candasm/MobileDetect",
+  "require": {
+    "yiisoft/yii": "1.1.14",
```
Comment (iamsalnikov) [inline composer.json:22]: "yiisoft/yii": "1.1.*"

File: src/MobileDetectComponent.php (hunk 2 above)
Comment (candasm) [reply src/MobileDetectComponent.php:7]: Yii1 does not support it so I didn't add it.

File: src/MobileDetectComponent.php (hunk 1 above)
Comment (candasm) [reply src/MobileDetectComponent.php:4]: Because of you didn't answer 2 year and I changed almost everything.

File: composer.json (hunk 5 above)
Comment (candasm) [reply composer.json:23]: I have no idea if mobiledetect will work with Yii1 series so I accepted 2.7.6 version.

File: composer.json (hunk 6 above)
Comment (candasm) [reply composer.json:22]: sure

File: src/MobileDetectComponent.php (hunk 2 above)
Comment (iamsalnikov) [reply src/MobileDetectComponent.php:7]: Yii1 does not support namespaces? Because minimum version of php is 5.1.0?

File: composer.json (hunk 5 above)
Comment (iamsalnikov) [reply composer.json:23]: Hard fixation version dependencies may create a conflict if a project that uses the package will have a dependence, which also requires MobileDetect, but another version. If possible, it is desirable to use *

File: composer.json (hunk 3 above)
Comment (candasm) [reply composer.json:17]: YiiFramework 1.x series does not support namespace's and composer autoloader so it's not important.

Comment (iamsalnikov) [reply composer.json:17]: What you mean in "YiiFramework 1.x series does not support namespace's"?

Comment (candasm) [reply composer.json:17]: [Example](https://github.com/yiisoft/yii/blob/master/framework/collections/CAttributeCollection.php) class of Yii 1.x framework It does not use namespaces so we don't need to set a namespace. But if you want it for YiiFramework 2 this should be in another version.

Comment (iamsalnikov) [reply composer.json:17]: This fact, that Yii 1 classes defined without namespaces, does not tell to us, that yii 1 does not support namespaces. All of classes in yii1 just placed in root namespace

**Pull request closed without merging**